src/details/ArborX_DetailsTreeNodeLabeling.hpp

```diff
@@ -0,0 +1,114 @@
+/****************************************************************************
+ * Copyright (c) 2017-2021 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#ifndef ARBORX_DETAILS_TREE_NODE_LABELING_HPP
+#define ARBORX_DETAILS_TREE_NODE_LABELING_HPP
+
+#include <ArborX_DetailsHappyTreeFriends.hpp>
+#include <ArborX_LinearBVH.hpp>
```

Comment:
Hadn't noticed you included that header.  This is a bit excessive.

Reply:
Will push force
```diff
diff --git a/src/details/ArborX_DetailsTreeNodeLabeling.hpp b/src/details/ArborX_DetailsTreeNodeLabeling.hpp
index a3efeff..f2d9896 100644
--- a/src/details/ArborX_DetailsTreeNodeLabeling.hpp
+++ b/src/details/ArborX_DetailsTreeNodeLabeling.hpp
@@ -13,7 +13,9 @@
 #define ARBORX_DETAILS_TREE_NODE_LABELING_HPP

 #include <ArborX_DetailsHappyTreeFriends.hpp>
-#include <ArborX_LinearBVH.hpp>
+#include <ArborX_Exception.hpp>
+
+#include <Kokkos_Core.hpp>

 namespace ArborX
 {

Reply:
It was used when I tested by constructed a full BVH hierarchy. Sure, it's not needed anymore.